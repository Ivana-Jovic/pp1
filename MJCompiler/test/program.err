ERROR 20:33:31,959 - 
Sintaksna greska na liniji 30
ERROR 20:33:31,960 - *******DesignatorStatement -Izvrsen oporavak do ; u liniji 29
