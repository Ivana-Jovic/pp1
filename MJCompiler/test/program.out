INFO  20:33:31,912 - Compiling source file: C:\Users\Ivana\Desktop\PP1\workspace\MJCompiler\test\program.mj
INFO  20:33:31,948 - #2 program
INFO  20:33:31,948 - #49 test302
INFO  20:33:31,948 - #7 const
INFO  20:33:31,949 - #49 int
INFO  20:33:31,949 - #49 jedan
INFO  20:33:31,949 - #35 =
INFO  20:33:31,949 - #48 1
INFO  20:33:31,949 - #39 ;
INFO  20:33:31,950 - #49 int
INFO  20:33:31,950 - #49 i
INFO  20:33:31,950 - #39 ;
INFO  20:33:31,950 - #49 bool
INFO  20:33:31,950 - #49 ibool
INFO  20:33:31,950 - #39 ;
INFO  20:33:31,950 - #49 int
INFO  20:33:31,950 - #49 niz2
INFO  20:33:31,950 - #39 ;
INFO  20:33:31,950 - #21 record
INFO  20:33:31,950 - #49 Test
INFO  20:33:31,951 - #42 {
INFO  20:33:31,951 - #49 int
INFO  20:33:31,951 - #49 niz
INFO  20:33:31,951 - #46 [
INFO  20:33:31,951 - #47 ]
INFO  20:33:31,951 - #39 ;
INFO  20:33:31,951 - #49 int
INFO  20:33:31,951 - #49 a
INFO  20:33:31,951 - #39 ;
INFO  20:33:31,952 - #43 }
INFO  20:33:31,952 - #49 Test
INFO  20:33:31,952 - #49 t1
INFO  20:33:31,952 - #40 ,
INFO  20:33:31,952 - #49 t2
INFO  20:33:31,952 - #39 ;
INFO  20:33:31,952 - #42 {
INFO  20:33:31,952 - #49 int
INFO  20:33:31,953 - #49 sum
INFO  20:33:31,953 - #44 (
INFO  20:33:31,953 - #49 int
INFO  20:33:31,953 - #49 z
INFO  20:33:31,953 - #40 ,
INFO  20:33:31,953 - #49 int
INFO  20:33:31,953 - #49 d
INFO  20:33:31,953 - #45 )
INFO  20:33:31,953 - #42 {
INFO  20:33:31,953 - #14 return
INFO  20:33:31,953 - #49 z
INFO  20:33:31,954 - #23 +
INFO  20:33:31,954 - #49 d
INFO  20:33:31,955 - #39 ;
INFO  20:33:31,955 - #43 }
INFO  20:33:31,955 - #49 int
INFO  20:33:31,956 - #49 inkrement
INFO  20:33:31,956 - #44 (
INFO  20:33:31,956 - #49 int
INFO  20:33:31,956 - #49 a
INFO  20:33:31,956 - #40 ,
INFO  20:33:31,956 - #49 int
INFO  20:33:31,956 - #49 inc
INFO  20:33:31,956 - #45 )
INFO  20:33:31,957 - #49 int
INFO  20:33:31,957 - #49 pom
INFO  20:33:31,957 - #39 ;
INFO  20:33:31,957 - #42 {
INFO  20:33:31,957 - #8 if
INFO  20:33:31,957 - #44 (
INFO  20:33:31,957 - #49 inc
INFO  20:33:31,957 - #27 ==
INFO  20:33:31,957 - #49 jedan
INFO  20:33:31,958 - #45 )
INFO  20:33:31,958 - #49 pom
INFO  20:33:31,959 - #35 =
INFO  20:33:31,959 - #48 1
INFO  20:33:31,959 - #6 else
INFO  20:33:31,959 - #8 if
INFO  20:33:31,959 - #44 (
INFO  20:33:31,959 - #49 inc
INFO  20:33:31,960 - #27 ==
INFO  20:33:31,960 - #48 2
INFO  20:33:31,960 - #45 )
INFO  20:33:31,960 - #49 pom
INFO  20:33:31,960 - #35 =
INFO  20:33:31,960 - #49 a
INFO  20:33:31,960 - #39 ;
INFO  20:33:31,960 - #14 return
INFO  20:33:31,960 - #49 sum
INFO  20:33:31,960 - #44 (
INFO  20:33:31,961 - #49 a
INFO  20:33:31,961 - #40 ,
INFO  20:33:31,961 - #49 pom
INFO  20:33:31,961 - #45 )
INFO  20:33:31,962 - #39 ;
INFO  20:33:31,962 - #43 }
INFO  20:33:31,962 - #49 bool
INFO  20:33:31,962 - #49 verify
INFO  20:33:31,962 - #44 (
INFO  20:33:31,962 - #45 )
INFO  20:33:31,962 - #42 {
INFO  20:33:31,962 - #12 print
INFO  20:33:31,962 - #44 (
INFO  20:33:31,962 - #50 v
INFO  20:33:31,962 - #45 )
INFO  20:33:31,962 - #39 ;
INFO  20:33:31,962 - #14 return
INFO  20:33:31,962 - #51 false
INFO  20:33:31,963 - #39 ;
INFO  20:33:31,963 - #43 }
INFO  20:33:31,963 - #15 void
INFO  20:33:31,963 - #49 main
INFO  20:33:31,963 - #44 (
INFO  20:33:31,963 - #45 )
INFO  20:33:31,963 - #42 {
INFO  20:33:31,963 - #8 if
INFO  20:33:31,963 - #44 (
INFO  20:33:31,963 - #48 1
INFO  20:33:31,963 - #28 !=
INFO  20:33:31,963 - #48 1
INFO  20:33:31,964 - #45 )
INFO  20:33:31,964 - #12 print
INFO  20:33:31,964 - #44 (
INFO  20:33:31,964 - #22 -
INFO  20:33:31,964 - #48 6
INFO  20:33:31,964 - #45 )
INFO  20:33:31,964 - #39 ;
INFO  20:33:31,964 - #6 else
INFO  20:33:31,964 - #12 print
INFO  20:33:31,965 - #44 (
INFO  20:33:31,965 - #48 0
INFO  20:33:31,965 - #45 )
INFO  20:33:31,965 - #39 ;
INFO  20:33:31,965 - #49 ibool
INFO  20:33:31,965 - #35 =
INFO  20:33:31,965 - #51 false
INFO  20:33:31,965 - #39 ;
INFO  20:33:31,965 - #43 }
INFO  20:33:31,965 - #43 }
INFO  20:33:31,967 - Program(
  ProgName(
   test302
  ) [ProgName]
  ProgListVarDecl(
    ProgListRecordDecl(
      ProgListVarDecl(
        ProgListVarDecl(
          ProgListVarDecl(
            ProgListConstDecl(
              ProgListNoProgList(
                Epsilon(
                ) [Epsilon]
              ) [ProgListNoProgList]
              ConstDeclNoError(
                Type(
                 int
                ) [Type]
                ConstTypeNCB(
                 jedan
                  NCBConstNum(
                   1
                  ) [NCBConstNum]
                ) [ConstTypeNCB]
                NoConstDeclList(
                  Epsilon(
                  ) [Epsilon]
                ) [NoConstDeclList]
              ) [ConstDeclNoError]
            ) [ProgListConstDecl]
            VarDeclSquare(
              Type(
               int
              ) [Type]
              VarDeclListNoLong(
                VariableNoArray(
                 i
                ) [VariableNoArray]
              ) [VarDeclListNoLong]
            ) [VarDeclSquare]
          ) [ProgListVarDecl]
          VarDeclSquare(
            Type(
             bool
            ) [Type]
            VarDeclListNoLong(
              VariableNoArray(
               ibool
              ) [VariableNoArray]
            ) [VarDeclListNoLong]
          ) [VarDeclSquare]
        ) [ProgListVarDecl]
        VarDeclSquare(
          Type(
           int
          ) [Type]
          VarDeclListNoLong(
            VariableNoArray(
             niz2
            ) [VariableNoArray]
          ) [VarDeclListNoLong]
        ) [VarDeclSquare]
      ) [ProgListVarDecl]
      RecordDecl(
        RecordDeclName(
         Test
        ) [RecordDeclName]
        YVarDeclListList(
          YVarDeclListList(
            NoVarDeclListList(
              Epsilon(
              ) [Epsilon]
            ) [NoVarDeclListList]
            VarDeclSquare(
              Type(
               int
              ) [Type]
              VarDeclListNoLong(
                VariableArray(
                 niz
                ) [VariableArray]
              ) [VarDeclListNoLong]
            ) [VarDeclSquare]
          ) [YVarDeclListList]
          VarDeclSquare(
            Type(
             int
            ) [Type]
            VarDeclListNoLong(
              VariableNoArray(
               a
              ) [VariableNoArray]
            ) [VarDeclListNoLong]
          ) [VarDeclSquare]
        ) [YVarDeclListList]
      ) [RecordDecl]
    ) [ProgListRecordDecl]
    VarDeclSquare(
      Type(
       Test
      ) [Type]
      VarDeclListLong(
        VarDeclListNoLong(
          VariableNoArray(
           t1
          ) [VariableNoArray]
        ) [VarDeclListNoLong]
        VariableNoArray(
         t2
        ) [VariableNoArray]
      ) [VarDeclListLong]
    ) [VarDeclSquare]
  ) [ProgListVarDecl]
  YMethodDeclList(
    YMethodDeclList(
      YMethodDeclList(
        YMethodDeclList(
          NoMethodDeclList(
            Epsilon(
            ) [Epsilon]
          ) [NoMethodDeclList]
          MethodDecl(
            MethodTypeNameType(
              Type(
               int
              ) [Type]
             sum
            ) [MethodTypeNameType]
            YFormParsList(
              FormParsNoSquareList(
                FormParsNoSquare(
                  Type(
                   int
                  ) [Type]
                 z
                ) [FormParsNoSquare]
                Type(
                 int
                ) [Type]
               d
              ) [FormParsNoSquareList]
            ) [YFormParsList]
            NoVarDeclListList(
              Epsilon(
              ) [Epsilon]
            ) [NoVarDeclListList]
            YStatementList(
              NoStatementList(
                Epsilon(
                ) [Epsilon]
              ) [NoStatementList]
              StatementSingle(
                SingleStatementRetExpr(
                  ExprAddop(
                    ExprTerm(
                      TermFactor(
                        Factor(
                          UnaryNoMinus(
                            Epsilon(
                            ) [Epsilon]
                          ) [UnaryNoMinus]
                          FactorDes(
                            DesignatorId(
                             z
                            ) [DesignatorId]
                          ) [FactorDes]
                        ) [Factor]
                      ) [TermFactor]
                    ) [ExprTerm]
                    AddopPlus(
                    ) [AddopPlus]
                    TermFactor(
                      Factor(
                        UnaryNoMinus(
                          Epsilon(
                          ) [Epsilon]
                        ) [UnaryNoMinus]
                        FactorDes(
                          DesignatorId(
                           d
                          ) [DesignatorId]
                        ) [FactorDes]
                      ) [Factor]
                    ) [TermFactor]
                  ) [ExprAddop]
                ) [SingleStatementRetExpr]
              ) [StatementSingle]
            ) [YStatementList]
          ) [MethodDecl]
        ) [YMethodDeclList]
        MethodDecl(
          MethodTypeNameType(
            Type(
             int
            ) [Type]
           inkrement
          ) [MethodTypeNameType]
          YFormParsList(
            FormParsNoSquareList(
              FormParsNoSquare(
                Type(
                 int
                ) [Type]
               a
              ) [FormParsNoSquare]
              Type(
               int
              ) [Type]
             inc
            ) [FormParsNoSquareList]
          ) [YFormParsList]
          YVarDeclListList(
            NoVarDeclListList(
              Epsilon(
              ) [Epsilon]
            ) [NoVarDeclListList]
            VarDeclSquare(
              Type(
               int
              ) [Type]
              VarDeclListNoLong(
                VariableNoArray(
                 pom
                ) [VariableNoArray]
              ) [VarDeclListNoLong]
            ) [VarDeclSquare]
          ) [YVarDeclListList]
          YStatementList(
            YStatementList(
              NoStatementList(
                Epsilon(
                ) [Epsilon]
              ) [NoStatementList]
              StatementSingle(
                SingleStatementIf(
                  ConditionDone(
                    ConditionNoList(
                      CondTermDone(
                        CondTermNoList(
                          CondFactList(
                            ExprTerm(
                              TermFactor(
                                Factor(
                                  UnaryNoMinus(
                                    Epsilon(
                                    ) [Epsilon]
                                  ) [UnaryNoMinus]
                                  FactorDes(
                                    DesignatorId(
                                     inc
                                    ) [DesignatorId]
                                  ) [FactorDes]
                                ) [Factor]
                              ) [TermFactor]
                            ) [ExprTerm]
                            RelopEqual(
                            ) [RelopEqual]
                            ExprTerm(
                              TermFactor(
                                Factor(
                                  UnaryNoMinus(
                                    Epsilon(
                                    ) [Epsilon]
                                  ) [UnaryNoMinus]
                                  FactorDes(
                                    DesignatorId(
                                     jedan
                                    ) [DesignatorId]
                                  ) [FactorDes]
                                ) [Factor]
                              ) [TermFactor]
                            ) [ExprTerm]
                          ) [CondFactList]
                        ) [CondTermNoList]
                      ) [CondTermDone]
                    ) [ConditionNoList]
                  ) [ConditionDone]
                  StatementSingle(
                    SingleStatementDesStmtASSIGNOP(
                      DesignatorStatementError(
                      ) [DesignatorStatementError]
                    ) [SingleStatementDesStmtASSIGNOP]
                  ) [StatementSingle]
                ) [SingleStatementIf]
              ) [StatementSingle]
            ) [YStatementList]
            StatementSingle(
              SingleStatementRetExpr(
                ExprTerm(
                  TermFactor(
                    Factor(
                      UnaryNoMinus(
                        Epsilon(
                        ) [Epsilon]
                      ) [UnaryNoMinus]
                      FactorDesParenAct(
                        DesignatorId(
                         sum
                        ) [DesignatorId]
                        ActParsWrap(
                          ActParsFirst(
                            Epsilon(
                            ) [Epsilon]
                          ) [ActParsFirst]
                          ActParsList(
                            ActParsNoList(
                              ExprTerm(
                                TermFactor(
                                  Factor(
                                    UnaryNoMinus(
                                      Epsilon(
                                      ) [Epsilon]
                                    ) [UnaryNoMinus]
                                    FactorDes(
                                      DesignatorId(
                                       a
                                      ) [DesignatorId]
                                    ) [FactorDes]
                                  ) [Factor]
                                ) [TermFactor]
                              ) [ExprTerm]
                            ) [ActParsNoList]
                            ExprTerm(
                              TermFactor(
                                Factor(
                                  UnaryNoMinus(
                                    Epsilon(
                                    ) [Epsilon]
                                  ) [UnaryNoMinus]
                                  FactorDes(
                                    DesignatorId(
                                     pom
                                    ) [DesignatorId]
                                  ) [FactorDes]
                                ) [Factor]
                              ) [TermFactor]
                            ) [ExprTerm]
                          ) [ActParsList]
                        ) [ActParsWrap]
                      ) [FactorDesParenAct]
                    ) [Factor]
                  ) [TermFactor]
                ) [ExprTerm]
              ) [SingleStatementRetExpr]
            ) [StatementSingle]
          ) [YStatementList]
        ) [MethodDecl]
      ) [YMethodDeclList]
      MethodDecl(
        MethodTypeNameType(
          Type(
           bool
          ) [Type]
         verify
        ) [MethodTypeNameType]
        NoFormParsList(
          Epsilon(
          ) [Epsilon]
        ) [NoFormParsList]
        NoVarDeclListList(
          Epsilon(
          ) [Epsilon]
        ) [NoVarDeclListList]
        YStatementList(
          YStatementList(
            NoStatementList(
              Epsilon(
              ) [Epsilon]
            ) [NoStatementList]
            StatementSingle(
              SingleStatementPrint(
                ExprTerm(
                  TermFactor(
                    Factor(
                      UnaryNoMinus(
                        Epsilon(
                        ) [Epsilon]
                      ) [UnaryNoMinus]
                      FactorChar(
                       v
                      ) [FactorChar]
                    ) [Factor]
                  ) [TermFactor]
                ) [ExprTerm]
              ) [SingleStatementPrint]
            ) [StatementSingle]
          ) [YStatementList]
          StatementSingle(
            SingleStatementRetExpr(
              ExprTerm(
                TermFactor(
                  Factor(
                    UnaryNoMinus(
                      Epsilon(
                      ) [Epsilon]
                    ) [UnaryNoMinus]
                    FactorBool(
                     false
                    ) [FactorBool]
                  ) [Factor]
                ) [TermFactor]
              ) [ExprTerm]
            ) [SingleStatementRetExpr]
          ) [StatementSingle]
        ) [YStatementList]
      ) [MethodDecl]
    ) [YMethodDeclList]
    MethodDecl(
      MethodTypeNameVoid(
       main
      ) [MethodTypeNameVoid]
      NoFormParsList(
        Epsilon(
        ) [Epsilon]
      ) [NoFormParsList]
      NoVarDeclListList(
        Epsilon(
        ) [Epsilon]
      ) [NoVarDeclListList]
      YStatementList(
        YStatementList(
          NoStatementList(
            Epsilon(
            ) [Epsilon]
          ) [NoStatementList]
          StatementSingle(
            SingleStatementIfElse(
              ConditionDone(
                ConditionNoList(
                  CondTermDone(
                    CondTermNoList(
                      CondFactList(
                        ExprTerm(
                          TermFactor(
                            Factor(
                              UnaryNoMinus(
                                Epsilon(
                                ) [Epsilon]
                              ) [UnaryNoMinus]
                              FactorNum(
                               1
                              ) [FactorNum]
                            ) [Factor]
                          ) [TermFactor]
                        ) [ExprTerm]
                        RelopNotEq(
                        ) [RelopNotEq]
                        ExprTerm(
                          TermFactor(
                            Factor(
                              UnaryNoMinus(
                                Epsilon(
                                ) [Epsilon]
                              ) [UnaryNoMinus]
                              FactorNum(
                               1
                              ) [FactorNum]
                            ) [Factor]
                          ) [TermFactor]
                        ) [ExprTerm]
                      ) [CondFactList]
                    ) [CondTermNoList]
                  ) [CondTermDone]
                ) [ConditionNoList]
              ) [ConditionDone]
              StatementSingle(
                SingleStatementPrint(
                  ExprTerm(
                    TermFactor(
                      Factor(
                        UnaryMinus(
                        ) [UnaryMinus]
                        FactorNum(
                         6
                        ) [FactorNum]
                      ) [Factor]
                    ) [TermFactor]
                  ) [ExprTerm]
                ) [SingleStatementPrint]
              ) [StatementSingle]
              ElseWrapper(
              ) [ElseWrapper]
              StatementSingle(
                SingleStatementPrint(
                  ExprTerm(
                    TermFactor(
                      Factor(
                        UnaryNoMinus(
                          Epsilon(
                          ) [Epsilon]
                        ) [UnaryNoMinus]
                        FactorNum(
                         0
                        ) [FactorNum]
                      ) [Factor]
                    ) [TermFactor]
                  ) [ExprTerm]
                ) [SingleStatementPrint]
              ) [StatementSingle]
            ) [SingleStatementIfElse]
          ) [StatementSingle]
        ) [YStatementList]
        StatementSingle(
          SingleStatementDesStmtASSIGNOP(
            DesignatorAssignop(
              DesignatorId(
               ibool
              ) [DesignatorId]
              Assignop(
              ) [Assignop]
              ExprTerm(
                TermFactor(
                  Factor(
                    UnaryNoMinus(
                      Epsilon(
                      ) [Epsilon]
                    ) [UnaryNoMinus]
                    FactorBool(
                     false
                    ) [FactorBool]
                  ) [Factor]
                ) [TermFactor]
              ) [ExprTerm]
            ) [DesignatorAssignop]
          ) [SingleStatementDesStmtASSIGNOP]
        ) [StatementSingle]
      ) [YStatementList]
    ) [MethodDecl]
  ) [YMethodDeclList]
) [Program]
INFO  20:33:31,968 - ===================================
INFO  20:33:31,972 - Parsiranje NIJE uspesno zavrseno
INFO  20:33:31,972 - GRESAKA U PARS:true
